core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -57,13 +57,11 @@ class LogSegment private[log] (val log: FileRecords,
                                val baseOffset: Long,
                                val indexIntervalBytes: Int,
                                val rollJitterMs: Long,
-                               val maxSegmentMs: Long,
-                               val maxSegmentBytes: Int,
                                val time: Time) extends Logging {
 
-  def shouldRoll(messagesSize: Int, maxTimestampInMessages: Long, maxOffsetInMessages: Long, now: Long): Boolean = {
-    val reachedRollMs = timeWaitedForRoll(now, maxTimestampInMessages) > maxSegmentMs - rollJitterMs
-    size > maxSegmentBytes - messagesSize ||
+  def shouldRoll(config: LogConfig, messagesSize: Int, maxTimestampInMessages: Long, maxOffsetInMessages: Long, now: Long): Boolean = {
```

Comment:
Hmm, why are we passing the whole config when we only need two parameters? One thing we should consider is introducing a `RollParams` case class to make this more maintainable. Generally, it makes it harder to test and understand if we pass these config classes with lots of params when we only need a subset.

Reply:
`RollParams` could have an `apply` that takes `AppendInfo` and `LogConfig` which could be used by `Log.maybeRoll`.

Reply:
@ijuma  Thanks for the review. Yes, we can add a RollParams class to hold the params.
Added a commit which implements suggested approach. Pls take take a look, when you get a chance.